packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -33,12 +34,15 @@ export const useSwitch = (props: SwitchProps, animationConfig?: AnimationConfig)
   } = props;
 
   const [checkedState, toggleCallback] = useAsToggleWithEvent(defaultChecked, checked, onChange);
-
   //Setting the initial position of the knob on track when page loads.
   React.useEffect(() => {
     if (isMobile) {
       Animated.timing(animation, {
-        toValue: checkedState ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2) : 0,
+        toValue: checkedState
+          ? isRTL
+            ? -(animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2))
```

Comment:
let's avoid computing in "IF" statements as much as possible. fine if it can't be avoided.

Reply:
moved out to computing once only.